src/python/pants/engine/fs_test.py

```diff
@@ -40,871 +37,711 @@
+    )
+    assert_path_globs(
+        rule_runner,
+        ["**/*.txt", "a/b/1.txt", "4.txt"],
+        expected_files=[
+            "4.txt",
+            "a/3.txt",
+            "c.ln/1.txt",
+            "d.ln/3.txt",
+            "a/b/1.txt",
+            "d.ln/b/1.txt",
+        ],
+        expected_dirs=FS_TAR_ALL_DIRS,
+    )
 
-        def prepare(project_tree):
-            link_path = os.path.join(project_tree.build_root, link)
-            dest_path = os.path.join(project_tree.build_root, dest)
-            relative_symlink(dest_path, link_path)
 
-        # Because the symlink does not escape, it should be ignored.
-        self.assert_walk_files([link], [], prepare=prepare)
+def test_path_globs_parent_link(rule_runner: RuleRunner) -> None:
+    setup_fs_test_tar(rule_runner)
+    assert_path_globs(
+        rule_runner,
+        ["c.ln/../3.txt"],
+        expected_files=["c.ln/../3.txt"],
+        expected_dirs=["c.ln", "c.ln/.."],
+    )
 
-    def test_walk_symlink_dead_nested(self) -> None:
-        link = "subdir/dead"
-        dest = "this_folder_does_not_exist/this_file_does_not_exist"
 
-        def prepare(project_tree):
-            link_path = os.path.join(project_tree.build_root, link)
-            dest_path = os.path.join(project_tree.build_root, dest)
-            relative_symlink(dest_path, link_path)
+def test_path_globs_symlink_escaping_errors(rule_runner: RuleRunner) -> None:
+    setup_fs_test_tar(rule_runner)
+    link = os.path.join(rule_runner.build_root, "subdir/escaping")
+    dest = os.path.join(rule_runner.build_root, "../../..")
+    relative_symlink(dest, link)
 
-        # Because the symlink does not escape, it should be ignored.
-        self.assert_walk_files([link], [], prepare=prepare)
+    exc_reg = r".*While expanding link.*subdir/escaping.*may not traverse outside of the buildroot"
+    with pytest.raises(Exception, match=exc_reg):
+        assert_path_globs(rule_runner, ["subdir/escaping"], expected_files=[], expected_dirs=[])
 
-    def test_walk_recursive_all(self) -> None:
-        self.assert_walk_files(
-            ["**"],
-            [
-                "4.txt",
-                "a/3.txt",
-                "a/4.txt.ln",
-                "a/b/1.txt",
-                "a/b/2",
-                "c.ln/1.txt",
-                "c.ln/2",
-                "d.ln/3.txt",
-                "d.ln/4.txt.ln",
-                "d.ln/b/1.txt",
-                "d.ln/b/2",
-            ],
-        )
 
-    def test_walk_ignore(self) -> None:
-        # Ignore '*.ln' suffixed items at the root.
-        self.assert_walk_files(
-            ["**"],
-            ["4.txt", "a/3.txt", "a/4.txt.ln", "a/b/1.txt", "a/b/2"],
-            ignore_patterns=["/*.ln"],
-        )
-        # Whitelist one entry.
-        self.assert_walk_files(
-            ["**"],
-            ["4.txt", "a/3.txt", "a/4.txt.ln", "a/b/1.txt", "a/b/2", "c.ln/1.txt", "c.ln/2"],
-            ignore_patterns=["/*.ln", "!c.ln"],
-        )
```

Comment:
I couldn't figure out how to port this test -- I don't think there's a mechanism to ignore an ignore.

Reply:
Yea, fine with removing this.